nav2_smac_planner/src/smac_planner_2d.cpp

```diff
@@ -250,28 +250,15 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
   // Compute plan
   Node2D::CoordinateVector path;
   int num_iterations = 0;
-  std::string error;
-  try {
-    if (!_a_star->createPath(
-        path, num_iterations, _tolerance / static_cast<float>(costmap->getResolution())))
-    {
-      if (num_iterations < _a_star->getMaxIterations()) {
-        error = std::string("no valid path found");
-      } else {
-        error = std::string("exceeded maximum iterations");
-      }
+  if (!_a_star->createPath(
+      path, num_iterations,
+      _tolerance / static_cast<float>(costmap->getResolution())))
+  {
+    if (num_iterations < _a_star->getMaxIterations()) {
+      throw nav2_core::NoValidPathCouldBeFound("no valid path found");
+    } else {
+      throw nav2_core::NoValidPathCouldBeFound("exceeded maximum iterations");
```

Comment:
This seems like `TIMEOUT` might be the right thing -- its not "time" out, but its out of valid iterations from a search-based planner

Reply:
@jwallace42 still relevant